nntrainer/models/neuralnet.cpp

```diff
@@ -629,6 +630,18 @@ int NeuralNetwork::train_run() {
 
   auto batch_size = std::get<props::TrainingBatchSize>(model_flex_props);
 
+  std::ofstream train_log_file;
+  auto &save_train_log_path =
+    std::get<props::SaveTrainLogPath>(model_flex_props);
+  if (!save_train_log_path.empty()) {
+    train_log_file =
+      checkedOpenStream<std::ofstream>(save_train_log_path, std::ios::out);
+  }
+
+  train_log_file << "progress : 0%" << "\n";
```

Comment:
you need to initialize `train_log_file` appropriately, for the case when `save_train_log_path` is not set by the user.
consider opening an ofstream to `\dev\null`.